pkg/utils/nodes.go

```diff
@@ -0,0 +1,25 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// isNodeNameValid returns an error if nodeName doesn't match any node name int the cluster, otherwise a nil
+func IsNodeNameValid(r client.Reader, nodeName string) error {
+	logger := ctrl.Log.WithName("utils-nodes")
+	node := &corev1.Node{}
+	objNodeName := types.NamespacedName{Name: nodeName}
+	err := r.Get(context.TODO(), objNodeName, node)
+	if err != nil {
+		errMsg := fmt.Sprintf("node %s is not part of the cluster's node names", nodeName)
```

Comment:
You need to check the error type. Only NotFound means that the node doesn't exist. There can be other temporary errors.